app/services/steam_importer_service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class SteamImporterService
+  class SteamImporterServiceFetchError < StandardError; end
+  class SteamImporterServiceUrlError < StandardError; end
+
+  def import_edition(url)
+    fetch_page(url)
+
+    edition = Edition.new
+    edition.title = title
+    edition.developer = developer
+    edition.publisher = publisher
+    edition.release_date = release_date
+    edition.description = description
+    edition.platform_id = platform
+    edition.region_id = region
+    edition.coverart = coverart
+    edition.media_id = media
+
+    edition
+  end
+
+  private
+    def fetch_page(url)
+      uri = URI(url)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      request = Net::HTTP::Get.new(uri)
+      # Cookies to bypass the age check for games with mature content
+      request["Cookie"] = "birthtime=596934001;lastagecheckage=1-0-1989;wants_mature_content=1"
+      response = http.request(request).body
+      @parsed_page = Nokogiri::HTML.parse(response)
+
+    rescue SocketError
+      raise SteamImporterServiceFetchError
+    rescue ArgumentError
```

Comment:
When does this happen?

Reply:
If the URL isn't in the correct format, it happens in the URI constructor